metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -78,7 +78,7 @@ final class MultilineStringFormattingProvider(
       newlineAdded: Boolean
   ): Boolean = {
     val newLineBeforePos =
-      text.lastIndexBetween('\n', upperBound = pos.start - 1)
```

Comment:
If I understand this correctly, this isn't needed here since we can just calculate starting at the position start rather then the position start -1 to determine the last index between the new line and the position start. This is what was throwing the exception when we were trying to get the index using the value here, which was -1 when in the beginning of the file.

Reply:
We probably just need `Math.max(pos.start - 1, 0)`

Reply:
I'm working on a fix for #1477, so I'll include the fix for both in another pr.

Reply:
When pressing enter we search for `newLineBeforePos`(where our newline was added). Then we search for '\n' before that and assign it to `pipeSearchStop` and it gives us a range [pipeSearchStop, newLineBeforePos] which is our previous line. Then we check if it contains '|' and if that is the case we do formatting(add '|' in current line properly aligned).

In current code it fails because it starts to search `pipeSearchStop` from position -1, I think we should simply add `Math.max(0, xxx - 1)` in both cases where we do search for \n

Reply:
@tgodzik you were faster with a comment, ignore mine :P